internal/handlers/jsonb1/where.go

```diff
@@ -142,6 +142,13 @@ func fieldExpr(field string, expr *types.Document, p *pg.Placeholder) (sql strin
 			// {field: {$gte: value}}
 			sql += "_jsonb->" + p.Next() + " >="
 			argSql, arg, err = scalar(value, p)
+		case "$size":
+			// { field: { $size: 2 } }
+			sql += "jsonb_array_length(_jsonb::jsonb->" + p.Next() + ") = "
+
+			var sizeValue any
+			sizeValue, err = expr.Get(expr.Command())
```

Comment:
```suggestion
			sizeValue, err = expr.Get(op)
```

But we should also check type

Reply:
I've updated the SQL query with type check, thank you!

Reply:
We should check that the `$size` operator is called on the array with integer (btw, can it be float?) argument, and return MongoDB-compatible errors if types are different. We should also have tests for those errors.

I think that invoking `jsonb_array_length` on the argument and checking the error code later might be easier than doing a type checking in SQL.

Reply:
1. I've added tests for `$size`.
2. Regarding invoking `jsonb_typeof()=array` in SQL. When I filled up the collection with mixed types values `jsonb_array_length()` function just started throwing exceptions for non-array values. At the same time, Mongo handles such cases silently skipping non-array values. So I think that type check is justified.

Reply:
> silently skipping non-array values

That what I was missing. Thanks.